src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/TwitterWidgetPageObject.java

```diff
@@ -46,26 +52,28 @@ protected String getErrorMessage() {
   }
 
   protected boolean isTagLoadedOnMercury() {
-    if(!isElementVisible(twitterIframe)) {
-      return false;
-    }
-
-    driver.switchTo().frame(twitterIframe);
-    boolean result = isElementVisible(twitterBody);
-    driver.switchTo().parentFrame();
-
-    return result;
+    return isWidgetVisible(twitterIframe, twitterBody);
   }
 
   protected boolean isTagLoadedOnOasis() {
-    if(!isElementVisible(twitterIframe)) {
-      return false;
-    }
+    return isWidgetVisible(twitterIframe, twitterBody);
+  }
 
-    driver.switchTo().frame(twitterIframe);
-    boolean result = isElementVisible(twitterBody);
-    driver.switchTo().parentFrame();
+  protected boolean areTagsLoadedOnOasis() {
```

Comment:
When I was writing this, I wanted to abstract almost everything… but I was afraid. :fearful: